base/cvd/cuttlefish/host/commands/assemble_cvd/flags.cc

```diff
@@ -1446,6 +1447,25 @@ Result<void> SetFlagDefaultsForVmm(
   return {};
 }
 
+Result<void> SetFlagDefaultsFromConfig() {
+  if (!FileExists(kDefaultsFilePath)) {
+    LOG(INFO) << "SetFlagDefaultsFromConfig: No flag defaults to override.";
+    return {};
+  }
+
+  std::string defaults_str;
+  CF_EXPECT(android::base::ReadFileToString(kDefaultsFilePath, &defaults_str),
+            "Couldn't read defaults file.");
+  std::map<std::string, std::string> defaults = CF_EXPECT(
+      ParseKeyEqualsValue(defaults_str), "Couldn't parse defaults file.");
```

Comment:
Cuttlefish uses the `key=value` format to report guest-build-specific information in the `android-info.txt` file. Recently however there was a [design](http://goto.google.com/cf-guest-config-pb) around moving to a textproto file, as another team felt constrained by the limitations of `key=value` settings alone.

If you can see a future where we might want to express more complex configurations, should we skip straight to a more structured data representation?

Reply:
I don't think there should be a need for more structured representation. We can take a policy perspective that gated features for rollout should be only binary and set logic governing any rollout rampup existing outside of the gated system itself.

Reply:
`ConfigFragment` was originally designed as an attempt to decompose the `CuttlefishConfig` class to stop being a a kitchen sink of different values, but I never followed through on it.